camus-sweeper/src/main/java/com/linkedin/camus/sweeper/CamusHourlySweeper.java

```diff
@@ -0,0 +1,327 @@
+package com.linkedin.camus.sweeper;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.Properties;
+import java.util.Map.Entry;
+import java.util.concurrent.Future;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.PathFilter;
+import org.apache.hadoop.mapreduce.lib.input.FileInputFormat;
+import org.apache.hadoop.mapreduce.lib.output.FileOutputFormat;
+import org.apache.hadoop.util.ToolRunner;
+import org.apache.log4j.Logger;
+
+import com.linkedin.camus.sweeper.mapreduce.CamusSweeperJob;
+import com.linkedin.camus.sweeper.utils.Utils;
+
+
+public class CamusHourlySweeper extends CamusSweeper {
+
+  private static final Logger LOG = Logger.getLogger(CamusHourlySweeper.class);
+
+  private static final String REDUCE_COUNT_OVERRIDE = "reduce.count.override.";
+  private static final String CAMUS_SWEEPER_TMP_DIR = "camus.sweeper.tmp.dir";
+  private static final String MAX_FILES = "max.files";
+  private static final int DEFAULT_MAX_FILES = 24;
+  private static final String TMP_PATH = "tmp.path";
+  private static final String CAMUS_SWEEPER_IO_CONFIGURER_CLASS = "camus.sweeper.io.configurer.class";
+  private static final String MAPRED_COMPRESS_MAP_OUTPUT = "mapred.compress.map.output";
+  private static final boolean DEFAULT_MAPRED_COMPRESS_MAP_OUTPUT = Boolean.TRUE;
+  private static final String REDUCER_COUNT = "reducer.count";
+  private static final int DEFAULT_REDUCER_COUNT = 45;
+  private static final String MAPRED_MIN_SPLIT_SIZE = "mapred.min.split.size";
+  private static final String MAPRED_MAX_SPLIT_SIZE = "mapred.max.split.size";
+
+  static final String TOPIC_AND_HOUR = "topic.and.hour";
+  static final String INPUT_PATHS = "input.paths";
+  static final String DEST_PATH = "dest.path";
+
+  private final CamusSweeperMetrics metrics;
+
+  public CamusHourlySweeper() {
+    super();
+    metrics = new CamusSweeperMetrics();
+  }
+
+  public CamusHourlySweeper(Properties props) {
+    super(props);
+    metrics = new CamusSweeperMetrics();
+  }
+
+  @Override
+  public Map<FileStatus, String> findAllTopics(Path input, PathFilter filter, String topicSubdir, FileSystem fs)
```

Comment:
Doesn't seem to need entire FileStatus, but this is overriding the superclass method, and I'm not planning to refactor superclass, which may create more bugs and need more testing.